### PR TITLE
fix bug: application can not create component which contains namespace resource

### PR DESCRIPTION
**Pull request by yangsoon**
**What this PR does / why we need it**:

Fix bug: Application can not create component which contains namespace resource.

Initializer rely on the app's ability to create resources, sometimes initializing an environment requires creating namespace,
this pr will fix the bug which application can not create namespace  resource.

### CODE REVIEW

File: vela-templates/addons/terraform/resource/namespace.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: tc-system
```
Comment (wonderflow) [inline vela-templates/addons/terraform/resource/namespace.yaml:4]:
```suggestion
  name: terraform-system
```

File: pkg/controller/core.oam.dev/v1alpha2/application/assemble/assemble.go [hunk 2]
```diff
@@ -300,6 +301,11 @@ func (am *AppManifests) filterAndSetAnnotations(obj *unstructured.Unstructured)
 }
 
 func (am *AppManifests) setNamespace(obj *unstructured.Unstructured) {
+	gvk := obj.GetObjectKind().GroupVersionKind()
+	if gvk == corev1.SchemeGroupVersion.WithKind(reflect.TypeOf(corev1.Namespace{}).Name()) {
```
Comment (wonderflow) [inline pkg/controller/core.oam.dev/v1alpha2/application/assemble/assemble.go:305]: please write a test for it

File: pkg/controller/core.oam.dev/v1alpha2/application/assemble/assemble.go [hunk 3]
```diff
@@ -300,6 +301,11 @@ func (am *AppManifests) filterAndSetAnnotations(obj *unstructured.Unstructured)
 }
 
 func (am *AppManifests) setNamespace(obj *unstructured.Unstructured) {
```
Comment (chivalryq) [inline pkg/controller/core.oam.dev/v1alpha2/application/assemble/assemble.go:303]: please add comment like “setNamespace set namespace except for Namespace object"

Comment (chivalryq) [reply pkg/controller/core.oam.dev/v1alpha2/application/assemble/assemble.go:303]: to make it clearer for other developer

Comment (yangsoon) [reply pkg/controller/core.oam.dev/v1alpha2/application/assemble/assemble.go:303]: got it, thanks

File: vela-templates/addons/terraform/resource/namespace.yaml (hunk 1 above)
Comment (yangsoon) [reply vela-templates/addons/terraform/resource/namespace.yaml]: got it

File: pkg/controller/core.oam.dev/v1alpha2/application/assemble/assemble.go (hunk 2 above)
Comment (yangsoon) [reply pkg/controller/core.oam.dev/v1alpha2/application/assemble/assemble.go:305]: got it

**Review by wonderflow — APPROVED** [review]

**Pull request merged**